sim-lib/src/lib.rs

```diff
@@ -35,6 +35,29 @@ pub enum NodeId {
     Alias(String),
 }
 
+impl NodeId {
+    pub fn get_pk(&self) -> Result<&PublicKey, String> {
+        if let NodeId::PublicKey(pk) = self {
+            Ok(pk)
+        } else {
+            panic!("NodeId is not a PublicKey. This should never happen, please report")
```

Comment:
maybe we could just return a `SimulationError`. No need to panic :)

Reply:
We can message this as `"NodeId is not a PublicKey. Please prefer to use actual node pubkey for the node aliased as {}"`. 

Any dependent on `sim-lib` who passes in activity definition with aliases knows what to change, _instead of reporting (?)_

Reply:
Ups, I forgot to change that to an error.

Regarding encouraging one or the other, I don't think we should. I'm happy with getting rid of the reporting stuff if we think of this as something that can be used by a third party, but we actually don't care whether they use one variant or the other, we cast everything to PK internally

Reply:
Fixed